src/agent/src/mount.rs

```diff
@@ -289,6 +290,82 @@ async fn virtio9p_storage_handler(
     common_storage_handler(logger, storage)
 }
 
+#[instrument]
+async fn hugetlbfs_storage_handler(
+    logger: &Logger,
+    storage: &Storage,
+    _sandbox: Arc<Mutex<Sandbox>>,
+) -> Result<String> {
+    // Allocate hugepages before mount
+    // /sys/kernel/mm/hugepages/hugepages-1048576kB/nr_hugepages
+    // /sys/kernel/mm/hugepages/hugepages-2048kB/nr_hugepages
+    // options eg "pagesize=2048,size=107374182"
+    allocate_hugepages(logger, &storage.options.to_vec()).context("allocate hugepages")?;
+
+    common_storage_handler(logger, storage)
+}
+
+// Allocate hugepages by writing to sysfs
+fn allocate_hugepages(logger: &Logger, options: &Vec<String>) -> Result<()> {
+    info!(logger, "mounting hugePages storage options: {:?}", options);
+
+    let (pagesize_str, size_str) = get_pagesize_and_size_from_option(options)?;
+    let pagesize = pagesize_str.parse::<i64>()?;
+    let size = size_str.parse::<i64>()?;
+
+    // sysfs entry is always of the form hugepages-${pagesize}kB
+    // Ref: https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt
+    let path = Path::new(SYS_FS_HUGEPAGES_PREFIX).join(format!("hugepages-{}kB", size / 1024));
+    fs::create_dir_all(&path).context(format!("create nr_hugepages directory {:?}", &path))?;
+
+    // write numpages to nr_hugepages file.
+    let path = path.join("nr_hugepages");
+    let numpages = format!("{}", size / pagesize);
+
+    let mut file = OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .open(&path)?;
+
+    file.write_all(numpages.as_bytes())
+        .context(format!("write nr_hugepages faile: {:?}", &path))?;
+
+    Ok(())
+}
+
+// Parse filesystem options string to retrieve hugepage details
+// options eg "pagesize=2048,size=107374182"
+fn get_pagesize_and_size_from_option(options: &Vec<String>) -> Result<(String, String)> {
+    let mut pagesize_str = "";
+    let mut size_str = "";
```

Comment:
An `Option` would be cleaner here. Something like:

```rust
let mut pagesize: Option<&str> = None;

   :

pagesize = Some(stripped);
```

That way you don't need `found` either as you could say:

```rust
if pagesize.is_none() || size.is_none() {
   return Err(anyhow!("no pagesize/size options found"));
}
```